solana/modules/wormhole-anchor-sdk/src/token_bridge/message.rs

```diff
@@ -22,35 +22,51 @@ pub struct TransferWithMeta {
 }
 
 impl AnchorDeserialize for TransferWithMeta {
-    fn deserialize(buf: &mut &[u8]) -> io::Result<Self> {
+    fn deserialize_reader<R: io::Read>(reader: &mut R) -> io::Result<Self> {
+        let mut variant = 0;
+        reader.read_exact(slice::from_mut(&mut variant))?;
         // Verify Payload ID is a token transfer with payload.
-        if buf[0] != PAYLOAD_ID_TRANSFER_WITH_PAYLOAD {
+        if variant != PAYLOAD_ID_TRANSFER_WITH_PAYLOAD {
             return Err(io::Error::new(
                 io::ErrorKind::InvalidInput,
                 "Invalid Token Bridge Transfer With Payload",
             ));
         }
 
+        // Skip the next 24 bytes:
+        {
+            let mut _discard = [0u8; 32];
+            reader.read_exact(&mut _discard)?;
+        }
+
         // Encoded amount should be the last 8 bytes of bytes 1 through 33,
         // otherwise we will have serious issues in the Token Bridge program.
         let amount = {
-            let mut out = [0u8; 8];
-            out.copy_from_slice(&buf[25..33]);
-            u64::from_be_bytes(out)
+            let mut buf = [0u8; 8];
+            reader.read_exact(&mut buf)?;
+            u64::from_be_bytes(buf)
```

Comment:
The reading of uints in this function should be DRYer. Implementing a generic function that leverages the BITS constant from uint types should work, I think.

Reply:
I'm using read now, as suggested below.